File: 008_fahrenheit_celsius/program.nim

import std/[math, strutils]

#[
iterator countup2*(a: float, b: float, step = 1.0): float {.inline.} =
  var res:float = a
  while res <= b:
    yield res
    res += step
]#

var
  #celsius: float
  fahr, celsius: float
  low, up, step: float

low = 0.0   # from 0
up = 300.0  # to 300
step = 20.0 # with step 20

echo "Fahrenheit   Celsius"

#for fahr in countup2(low, up, step):
while fahr <= up:
  celsius = (5.0 / 9.0) * (fahr - 32.0)
  echo fahr, "   ", celsius.formatFloat(ffDecimal, 1)
  fahr = fahr + step

echo "done!"

File: 008_fahrenheit_celsius/program.c

#include <stdio.h>

int main(void) {
  // fahr is 0, 20, ... 300
  
  float fahr, celsius;
  int low, up, step;
  
  low = 0;   // from 0
  up = 300;  // to 300
  step = 20; // with step 20
  
  printf("Fahrenheit   Celsius\n");
  
  for(fahr = low; fahr <= up; fahr = fahr + step){
    celsius = (5.0 / 9.0) * (fahr - 32.0);
    
    printf("%3.0f   %6.1f\n", fahr, celsius);
  }
  
  printf("done!\n");
  
  return 0;
}


